simulator_service/src/simulator_service_node.cpp

```diff
@@ -0,0 +1,91 @@
+#include <ros/ros.h>
+
+#include <simulator_service/SimulateTrajectory.h>
+
+#include <actionlib/client/simple_action_client.h>
+#include <control_msgs/FollowJointTrajectoryAction.h>
+
+static void scaleDurations(trajectory_msgs::JointTrajectory& traj, double scale)
+{
+  for (std::size_t i = 0; i < traj.points.size(); ++i)
+  {
+    traj.points[i].time_from_start = traj.points[i].time_from_start * scale;
+  }
+}
+
+namespace simulator_service
+{
+  class SimulatorService
+  {
+  public:
+    SimulatorService(ros::NodeHandle& nh, const std::string& service_name)
+      : ac_("joint_trajectory_action", true)
```

Comment:
This hard codes the name of the `joint_trajectory_action` server. Suggest to make this configurable.